[doc] update Thread Bourder Router documentation (#5443)
nrfconnect/sdk-nrf: doc/nrf/ug_thread_tools.rst

To program the nRF device with the RCP application, complete the following steps:

#. Build the :ref:`ot_coprocessor_sample` sample for the hardware platform and the transport of your choice:

   .. tabs::

      .. tab:: nRF52840 Dongle (USB transport)

         .. code-block:: console

>             west build -b nrf52840dongle_nrf52840 -- -DOVERLAY_CONFIG="overlay-rcp.conf ../common/overlay-thread_1_2_mtd.conf overlay-usb.conf"

Reviewer: Why the MTD overlay? Shouldn't be FTD for complete functionality? Or doesn't even matter for RCP?

Author: In general it should be the same since RCP provides only radio capabilities and the rest is depended on the host configuration. But with mtd overlay RCP image has less memory footprint.

Reviewer: But there could be dependencies, like FTD required for CSL transmitter, for instance.

>             west build -b nrf52840dongle_nrf52840 -- -DOVERLAY_CONFIG="overlay-rcp.conf ../common/overlay-thread_1_2_ftd.conf overlay-usb.conf"

Reviewer 2: nit: I think that `overlay-usb.conf` could be automatically included when building for `nrf52840dongle_nrf52840` but this is not related to this PR.

To set up and configure the OpenThread Border Router, follow the official `OpenThread Border Router Codelab tutorial`_ on the OpenThread documentation portal.
Omit the *Build and flash RCP firmware* section, because this section duplicates the steps performed in the previous section.

.. note::

   OTBR is developing every day and it may happen that the newest version is incompatible with the specific version of NCS.

>    To make sure that everything is working correctly use specific commit of OTBR which was tested with the right version of NCS.

Reviewer 3: ```suggestion
   To make sure that everything is working correctly, use a specific commit of OTBR that has been tested with the corresponding version of NCS.
```

To set up and configure the OpenThread Border Router, follow the official `OpenThread Border Router Codelab tutorial`_ on the OpenThread documentation portal.
Omit the *Build and flash RCP firmware* section, because this section duplicates the steps performed in the previous section.

.. note::

   OTBR is developing every day and it may happen that the newest version is incompatible with the specific version of NCS.
   To make sure that everything is working correctly, use a specific commit of OTBR that has been tested with the corresponding version of NCS.

   +--------------------------+--------------------+
   | OpenThread Border Router |        NCS         |
   +==========================+====================+

>    |        e149a60eb         |       v1.7         |

Reviewer: ```suggestion
   |        e149a60e          |       v1.7         |
```
Maybe use an even number of characters for the SHA?

Author: Why do you want to use event number?

Reviewer: Never mind, just realized that GitHub itself is using 7 chars as abbreviation.

>    |         e149a60          |       v1.7         |

Reviewer 2: I would prefer to have a simple command instead of the table. I don't think we need to maintain the table with versions for all releases anyway right? Users can switch between documentation of different releases.

So just have a sentence that: when following the CodeLab tutorial, after cloning the repository please ensure to checkout to the compatible commit id.
`git chekout e149a60`

      sudo docker network create --ipv6 --subnet fd11:db8:1::/64 -o com.docker.network.bridge.name=otbr0 otbr

> #. Download the compatible version of the OpenThread Bourder Router docker image by running the following command:

Reviewer 2: typo in the last commit: Bourder

      sudo docker network create --ipv6 --subnet fd11:db8:1::/64 -o com.docker.network.bridge.name=otbr0 otbr

> #. Download the latest version of the OpenThread Bourder Router docker container for nRF Connect SDK by running the following command:

Reviewer 4: 1. Typo: "Bourder"
2. It's no longer "the latest" - "compatible with this version of |NCS|"?
3. We don't download a container, we download an image based on which a container is created.

#. Build the applications with the required options.
   For example, to build the ``ot-cli`` application with support for USB transport and Thread v1.1, run the following command::

      ./script/cmake-build posix -DOT_THREAD_VERSION=1.1

   Alternatively, to build the ``ot-daemon`` and ``ot-ctl`` applications with support for Thread v1.2, run the following command::

      ./script/cmake-build posix -DOT_THREAD_VERSION=1.2 -DOT_DAEMON=ON

You can find the generated applications in :file:`./build/posix/src/posix/`.

Running the OpenThread POSIX applications
=========================================

Use the following radio URL parameter to connect to an RCP node.

* For UART transport:

  .. parsed-literal::
     :class: highlight

     'spinel+hdlc+uart://\ *ncp_uart_device*\ ?uart-baudrate=\ *baud_rate*'

>    Replace the following parameters:

Author: @wiba-nordic could you fix indentation for these changes? Formatting is still not working for me.
